ports/rp2/machine_uart.c

```diff
@@ -404,6 +404,11 @@ STATIC mp_uint_t machine_uart_read(mp_obj_t self_in, void *buf_in, mp_uint_t siz
     uint64_t timeout_char_us = (uint64_t)self->timeout_char * 1000;
     uint8_t *dest = buf_in;
 
+    // Force a few incoming bytes to the buffer
+    self->read_lock = true;
+    uart_drain_rx_fifo(self);
+    self->read_lock = false;
```

Comment:
Might this lead to lost characters, if the ringbuf is already full and there are bytes in the FIFO?

Reply:
I'll change that.